rules-reviewed/openshift/session.windup.xml

```diff
@@ -0,0 +1,87 @@
+                    </message>
+                    <link href="https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.0/html/development_guide/clustering_in_web_applications" title="JBoss EAP Clustering in Web Applications"/>
+                    <link href="https://access.redhat.com/documentation/en-us/red_hat_jboss_middleware_for_openshift/3/html-single/red_hat_jboss_enterprise_application_platform_for_openshift/#clustering" title="JBoss EAP Clustering for OpenShift"/>
+                    <tag>clustering</tag>
+                </hint>
+            </perform>
+        </rule>
+
+        <rule id="cloud-session-00001">
+            <when>
+                <javaclass references="javax.servlet.http.HttpSession.setAttribute({*})">
+                    <location>METHOD_CALL</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Session usage" effort="3" category-id="openshift-optional">
+                    <message>
+                    Detect storage of custom data in Http session by 'javax.http.HttpSession.setAttribute(...)'.  
+                    Consider that a cloud-enabled application must be stateless.  
+                    If required, session data must be distributed across all instances using a suitable caching framework.  
+                    Anyway, it is also possible to handle session stickiness in OpenShift  
+                    </message>
+                    <link href="https://access.redhat.com/documentation/en-us/red_hat_jboss_middleware_for_openshift/3/html/red_hat_jboss_data_grid_for_openshift/" title="Red Hat JBoss Data Grid for OpenShift"/>
+                    <link href="https://docs.openshift.com/container-platform/3.4/architecture/core_concepts/routes.html#haproxy-template-router" title="OpenShift: HAProxy Template Router - Sticky Session"/>
+                    <tag>clustering</tag>
+                </hint>
+            </perform>
+        </rule>
+
+        <rule id="cloud-session-00002">
+            <when>
+                <javaclass references="javax.servlet.http.HttpSession.putValue({*})">
+                    <location>METHOD_CALL</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Session usage" effort="3" category-id="openshift-optional">
+                    <message>
+                    Detect storage of custom data in Http session by 'javax.http.HttpSession.putValue(...)'.  
+                    Consider that a cloud-enabled application must be stateless.  
+                    If required, session data must be distributed across all instances using a suitable caching framework.  
+                    In addition, the 'putValue()' method is deprecated.  
+                    Anyway, it is also possible to handle session stickiness in OpenShift  
+                    </message>
```

Comment:
we can use markdown syntax in message text and if you use like CDATA it is safer.